test/integration/autonomous_upstream.cc

```diff
@@ -0,0 +1,75 @@
+#include "test/integration/autonomous_upstream.h"
+
+namespace Envoy {
+
+namespace {
+
+void HeaderToInt(const char header_name[], int32_t& return_int, Http::TestHeaderMapImpl& headers) {
+  std::string header_value = headers.get_(header_name);
+  if (!header_value.empty()) {
+    return_int = atoi(header_value.c_str());
```

Comment:
Should we use StringUtil::atoul here though and verify/assert that the header is a valid int?